### ISSUE TITLE
`_parse_expression` in `Leaf` class

### ISSUE THREAD

**Issue by lostptr** [bitbrain/beehave#227]
Godot version: 4.1.1

**Describe the bug**
Not really a bug, just something weird in the code.
There is expression parsing logic in the `Leaf` class that I think should not be part of this class.
ie. If I am making a RunAction what is the need to have the `_parse_expression` method? This method is only being used in the nodes used for unit testing and shoul not be part of the core library, at least not in the `Leaf` class.

**Screenshots**
![image](https://github.com/bitbrain/beehave/assets/21160216/01e1f9df-4408-4338-a022-12d7d2262477)

**Comment by bitbrain** [comment]
This is used for conditions: https://github.com/bitbrain/beehave/blob/godot-4.x/addons/beehave/nodes/leaves/blackboard_has.gd#L8

but it might be worth to refactor it so it does not clutter the Leaf node.

**Comment by lostptr** [comment]
After looking into the code, I cant think of a satisfying, "separation of concerns" way of doing this.
![image](https://github.com/bitbrain/beehave/assets/21160216/91a4940d-8b87-4cf6-be7a-ba097b290255)

I could create a "ExpressionParsingLeaf" between `Leaf` and `ActionLeaf` and `ConditionLeaf` but still I would not want those two classes to inherit that logic. The perfect solution for me was to have a class that BlackboardSet/Erase/Compare/HasCondition could inherit from but it is not possible. 

TLDR; meh, could not think of a much better alternative so I ok with leaving the way it is now 🤷‍♂️

**Comment by bitbrain** [comment]
We can improve this once traits are implemented in GDScript.
